BasicDateTimeTestSuite reworked

AdamBorek opened the pull request:
Facts have been broken up into multiple smaller ones

AdamBorek committed:
BasicDateTimeTestSuite reworked

Facts have been broken up into multiple smaller ones

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -1,216 +1,494 @@
 		output = date_inc_month(input, 11);
-		assert_equals(date_get_month(output), 8, "#2.2 date_inc_month( 11 ) overflow number of months");
+		assert_equals(date_get_month(output), 8, "date_inc_month( 11 ) overflow number of months");
 		assert_equals(date_get_year(output), date_get_year(input) + 1, "#2.2.1 date_inc_month( 11 ), should go to next year");
```
```suggestion

```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -1,216 +1,494 @@
-		assert_equals(date_get_hour(output), 15, "#4.2 date_inc_hour( 30 ) overflow number of hours");
-		assert_equals(date_get_day(output), date_get_day(input) + 1, "#4.2.1 date_inc_hour( 30 ), should go to next day");
+		assert_equals(date_get_hour(output), 15, "date_inc_hour( 30 ) overflow number of hours");
+		assert_equals(date_get_day(output), date_get_day(input) + 1, "date_inc_hour( 30 ), should go to next day");
```
```suggestion

```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -1,216 +1,494 @@
-		assert_equals(date_get_minute(output), 0, "#5.2 date_inc_minute( 17 ) overflow number of nimutes");
-		assert_equals(date_get_hour(output), date_get_hour(input) + 1, "#5.2.1 date_inc_minute( 17 ), should go to next hour");
+		assert_equals(date_get_minute(output), 0, "date_inc_minute( 17 ) overflow number of nimutes");
+		assert_equals(date_get_hour(output), date_get_hour(input) + 1, "date_inc_minute( 17 ), should go to next hour");
```
```suggestion

```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -224,19 +502,49 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
+	addFact("date_span_test #1", function() {
 
 		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
 		math_set_epsilon(0.00001)
```
```suggestion
		math_set_epsilon(0.00001);
```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
+	addFact("date_span_test #2", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
```suggestion
		math_set_epsilon(0.00001);
```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
+	addFact("date_span_test #3", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
```suggestion
		math_set_epsilon(0.00001);
```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
+	addFact("date_span_test #4", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
```suggestion
		math_set_epsilon(0.00001);
```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
+	addFact("date_span_test #5", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
```suggestion
		math_set_epsilon(0.00001);
```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
+	addFact("date_span_test #6", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
```suggestion
		math_set_epsilon(0.00001);
```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
+	addFact("date_span_test #7", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
```suggestion
		math_set_epsilon(0.00001);
```

Indibulous commented on projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
 		var old = date_get_timezone();
 
 		date_set_timezone(timezone_local);
 		assert_equals(date_get_timezone(), timezone_local, "#1 date_set_timezone( timezone_local ), failed to switch timezones");
```
```suggestion
		assert_equals(date_get_timezone(), timezone_local, "date_set_timezone( timezone_local ), failed to switch timezones");
```

Indibulous committed:
Apply suggestions from code review

Added some missing semicolons and fixed a small typo in one of the assert descriptions

Indibulous reviewed (approved):
Seems good! Added a few semicolons and fixed a typo - also marked a few lines that i think shouldn't be there? They look like they have been copied into the test below but forgot to be removed from the tests they are in.

Also, not super important, but might be worth removing all the comments with the old test numbers.

DiasFranciscoA commented:
@AdamBorek probably better to clean the old comments out of them to avoid misleading.. on of the reasons for the descriptions in the tests is so that you can do a CTRL + SHIFT + F and find the place in the code where the assert failed.. for easier debugging. so removing unnecessary comments is good. Also when inside loops try to make sure the loop index is some where in the description of the assert this way it's possible to know what iterations of the loop is causing the issue 🤔
